### DISCUSSION TITLE
2024-01-11

### DISCUSSION THREAD

**Discussion by yutingzhao1991** [ant-design/ant-design-web3#480] [中文社区双周会]
请回复待讨论的话题并记录相关纪要。

**Comment by yutingzhao1991** [comment]
上次会议待跟进事项进展：https://github.com/ant-design/ant-design-web3/discussions/411#discussioncomment-7963186

**Reply by yutingzhao1991** [reply]
<img width="881" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/1061968/64e2c4dd-66ff-4785-a9da-545b853e8a9c">

1、4 继续跟进，下次双周会之前完成。其它的已完成，具体见后面的纪要。

**Comment by yutingzhao1991** [comment]
讨论提案：
- 关于 wagmi 2.0 升级的问题 https://github.com/ant-design/ant-design-web3/issues/461
- 加密支付组件 https://github.com/ant-design/ant-design-web3/issues/415

**Comment by yutingzhao1991** [comment]
连接钱包交互优化方案 review： https://www.yuque.com/ant-design/pxl7ot/oee2i168dtri5aft

**Comment by yutingzhao1991** [comment]
ConnectButton size=small 情况在的 avatar 的问题也讨论下： https://github.com/ant-design/ant-design-web3/pull/387

**Reply by yutingzhao1991** [reply]
这个会和新的设计稿一起给出来

**Comment by gin-lsl** [comment]
多种链混合使用时的交互处理逻辑。

例如：
<img width="293" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/18319675/a5906b52-810b-41ad-9db9-0ea7ce6c65c4">

目前 @ant-design/web3-solana 中的逻辑是：
切换到 Ethereum 时，直接断开链接；切换到 Solana 时，再重新链接。

**Reply by yutingzhao1991** [reply]
这个我之前也想过。之前 wagmi 的 adapter 其实只支持以太坊兼容的链直接的切换。要做到不同的 adapter 的切换要思考一下。

solana adapter 的实现可以先不用考虑这个。

目前我的大概的想法是在 Web3ConfigProvider 这一层来支持。大概逻辑就是支持多个 adapter 的 Web3ConfigProvider 配置的合并。细节比较复杂。

**Reply by yutingzhao1991** [reply]
你现在这个逻辑如何实现的？

**Reply by Likang0122** [reply]
演示demo我觉得做成这样就够了，至少我没见过会同时使用两种链的dapp，要考虑到开发者的使用情况，不同的链的合约或者数据互相映射是一件非常麻烦的事情😝

**Reply by gin-lsl** [reply]
目前的处理逻辑是，因为 Solana 不像以太坊那样有很多链，所以我在 switchChain 里面判断如果不是 Solana，就直接断开链接。

**Comment by Likang0122** [comment]
还有一个就是，一个@gin-lsl 之前提供的想法，我觉得似乎可以做
做一个组件，支持主链币和标准代币的交易，或者是NFT或许也可以

**Reply by Likang0122** [reply]
大概就是<Transaction to={toAddress} value={'0.01'} /> 这样子的，展示上暂时没有想好怎么做合适些

**Reply by yutingzhao1991** [reply]
和 加密支付组件 https://github.com/ant-design/ant-design-web3/issues/415 一起考虑 @Likang0122

**Comment by yutingzhao1991** [comment]
会议纪要：
- 下次双周会之前 @wenqing993 这边给出 https://www.yuque.com/ant-design/pxl7ot/oee2i168dtri5aft 需求的 1.1、1.2、2.1 部分的设计稿。
- 下次双周会之前 @gin-lsl 完成 solana 链的 adapter 的开发
- @Likang0122 跟进支付组件的调研，下次会议对一下
- @Likang0122 列出具体要添加的 icon 到 issue 中， @wenqing993 那边可以先给 @yutingzhao1991  现在手里面已经有的 icon 加进去
